### PR TITLE
tagging for other stuff

### PR DESCRIPTION
**Pull request by amueller**

### CODE REVIEW

File: openml/tasks/task.py [hunk 1]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
 
         return self.split.repeats, self.split.folds, self.split.samples
 
+    def push_tag(self, tag):
+        """Annotates this flow with a tag on the server.
```
Comment (mfeurer) [inline openml/tasks/task.py]: Should be `Annotates thi task``, right?

File: openml/tasks/task.py [hunk 2]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+
+        Parameters
+        ----------
+        tag : str
+            Tag to attach to the flow.
```
Comment (mfeurer) [inline openml/tasks/task.py]: Same here.

File: openml/tasks/task.py [hunk 3]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+        """
+        data = {'flow_id': self.flow_id, 'tag': tag}
+        _perform_api_call("/flow/tag", data=data)
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: openml/tasks/task.py [hunk 4]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
 
+    def push_tag(self, tag):
+        """Annotates this flow with a tag on the server.
+
+        Parameters
+        ----------
+        tag : str
+            Tag to attach to the flow.
+        """
+        data = {'flow_id': self.flow_id, 'tag': tag}
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: openml/tasks/task.py [hunk 5]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+        _perform_api_call("/flow/tag", data=data)
+
+    def remove_tag(self, tag):
+        """Removes a tag from this flow on the server.
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: openml/tasks/task.py [hunk 6]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+
+        Parameters
+        ----------
+        tag : str
+            Tag to attach to the flow.
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: openml/tasks/task.py [hunk 7]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+        """
+        data = {'flow_id': self.flow_id, 'tag': tag}
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: openml/tasks/task.py [hunk 8]
```diff
@@ -96,6 +96,28 @@ def get_split_dimensions(self):
...
+        _perform_api_call("/flow/untag", data=data)
```
Comment (mfeurer) [inline openml/tasks/task.py]: And here.

File: tests/test_flows/test_flow.py [hunk 9]
```diff
@@ -67,6 +67,19 @@ def test_get_flow(self):
         self.assertEqual(subflow_3.parameters['L'], '-1')
         self.assertEqual(len(subflow_3.components), 0)
 
+    def test_tagging(self):
+        flow = openml.flows.get_flow(4024)
```
Comment (mfeurer) [inline tests/test_flows/test_flow.py]: That's dangerous. If the server is purged, this flow will no longer exist. Maybe you should just query the server for a flow and take the first.

File: openml/tasks/task.py (hunk 8 above)
Comment (amueller) [reply openml/tasks/task.py]: looks like I missed one search and replace.

File: tests/test_flows/test_flow.py (hunk 9 above)
Comment (amueller) [reply tests/test_flows/test_flow.py]: This flow is used in another test, which is why I selected it ;) We can have the setup query for a flow and then use that everywhere, though.

Comment (amueller) [reply tests/test_flows/test_flow.py]: Though the server was purged and it didn't fail, right?

Comment (mfeurer) [reply tests/test_flows/test_flow.py]: But that flow is loaded from the production server.

Comment (mfeurer) [reply tests/test_flows/test_flow.py]: Querying for a flow would be better.

Comment (amueller) [reply tests/test_flows/test_flow.py]: lol it's actually using the production server in the other test...

**Comment by amueller** [comment]
should be good now

**Pull request merged**